stores/sql/main.go

```diff
@@ -191,6 +191,24 @@ func ArchiveContract(ctx context.Context, tx sql.Tx, fcid types.FileContractID,
 	if err != nil {
 		return fmt.Errorf("failed to delete contract_sectors: %w", err)
 	}
+
+	// delete host sectors that are no longer associated with any contract
+	_, err = tx.Exec(ctx, `DELETE FROM host_sectors
+WHERE NOT EXISTS (
+    SELECT 1
+    FROM contract_sectors
+    WHERE contract_sectors.db_sector_id = host_sectors.db_sector_id
+)
+OR NOT EXISTS (
+    SELECT 1
+    FROM contracts
+    INNER JOIN hosts ON contracts.host_id = hosts.id
+    WHERE contracts.archival_reason IS NULL
+    AND hosts.id = host_sectors.db_host_id
+)`)
```

Comment:
This query won't be correct once we upload to temporary storage. Because it will remove sectors that aren't part of any contract. There is going to be a gap between uploading a sector and appending it to a contract once we use RHP4 so we don't want to delete it during that time.

The only time we will need to delete a host sector is if no `contract_sector` exists after 72 hours of uploading. Because that means it wasn't pinned in time. So we probably need to add a `uploaded_at` unix timestamp to this table as well which we set to the time we last upserted it.

Reply:
I added `updated_at`, I think that's more in line with the `created_at` dates we already have in our database.